### DISCUSSION TITLE
now() returning current dateTime?

### DISCUSSION THREAD

**Discussion by HFolkertH** [cincheo/daquota#32] [Q&A] [answered]
Hi Renaud

My date validatiion  will not check if a given date is equal to today or later. I use now() for the today date.
Is Now() returning date or datetime?
26/5/22 should be okay, but I get a false.

![image](https://user-images.githubusercontent.com/22573455/170439321-1334796e-5cfe-4786-85f1-fc5103d8c087.png)

kind regards
Folkert

**Comment by renaudpawlak** [comment, marked as answer]
Manipulating dates in JS can be quite tricky. That's why DLite embeds the momentjs lib by default. You can check the doc here: https://momentjs.com/

For your question, ``now()`` is the same as ``new Date()``. So, to transform it in a timestamp, you can use the ``now().getTime()`` function. If you want to get a timestamp from a moment date, you can use ``moment().toDate().getTime()``. So, a solution to your problem is:

```javascript
moment($d('datepicker-0')).toDate().getTime() >= now().getTime()
```

However, the best way to do it would be to use the moment API, which would also allows you to have some granularity in the comparison (instead of using milliseconds timestamp).

For instance, to compare dates with a "day" granularity, just use :

```javascript
moment($d('datepicker-0')).isAfter(now(), 'day'); 
```
